packages/connect-form/src/components/advanced-options-tabs/authentication-tab/authentication-gssapi.tsx

```diff
@@ -1,9 +1,105 @@
 import React from 'react';
+import { Checkbox, TextInput } from '@mongodb-js/compass-components';
+
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+import { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import FormFieldContainer from '../../form-field-container';
+import {
+  ConnectionFormError,
+  errorMessageByFieldName,
+} from '../../../utils/validation';
+import {
+  getConnectionStringUsername,
+  parseAuthMechanismProperties,
+} from '../../../utils/connection-string-helpers';
+
+function AuthenticationGSSAPI({
+  errors,
+  connectionStringUrl,
+  updateConnectionFormField,
+}: {
+  connectionStringUrl: ConnectionStringUrl;
+  errors: ConnectionFormError[];
+  updateConnectionFormField: UpdateConnectionFormField;
+}): React.ReactElement {
+  const kerberosPrincipalError = errorMessageByFieldName(
+    errors,
+    'kerberosPrincipal'
+  );
+  const principal = getConnectionStringUsername(connectionStringUrl);
+  const authMechanismProperties =
+    parseAuthMechanismProperties(connectionStringUrl);
+  const serviceName = authMechanismProperties.get('SERVICE_NAME');
+  const serviceRealm = authMechanismProperties.get('SERVICE_REALM');
+  const canonicalizeHostname = authMechanismProperties.get(
+    'CANONICALIZE_HOST_NAME'
+  );
 
-function AuthenticationGSSAPI(): React.ReactElement {
   return (
     <>
-      <p>Kerberos</p>
+      <FormFieldContainer>
+        <TextInput
+          onChange={({
+            target: { value },
+          }: React.ChangeEvent<HTMLInputElement>) => {
+            updateConnectionFormField({
+              type: 'update-username',
+              username: value,
+            });
+          }}
+          label="Principal"
+          errorMessage={kerberosPrincipalError}
+          state={kerberosPrincipalError ? 'error' : undefined}
+          value={principal || ''}
```

Comment:
Previously we had an info link to https://docs.mongodb.com/manual/core/kerberos/#principals
on this input:
<img width="480" alt="Screen Shot 2022-01-25 at 3 57 22 PM" src="https://user-images.githubusercontent.com/1791149/151058480-aaa1a065-bfc4-47b0-8d41-7cb6f7fd1b9a.png">
Do we want to keep it?

Reply:
Ah that's a good one, yeah, i'll merge this and add it right away